File: daily/2026/2078_Two_Furthest_Houses_With_Different_Colors.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int maxDistance(vector<int>& colors) {
        int n = colors.size();
        if (colors[0] != colors[n-1]) return n-1;
        int ans = -1;
        for (int i = 1; i < n - 1; i++){
            if (colors[i] != colors[0]){
                ans = max({ans, i, n - 1 - i});
            }
        }
        return ans;
    }
};

int main(){
    vector<int> colors = {1,1,1,6,1,1,1};
    // vector<int> colors = {1,8,3,8,3};
    // vector<int> colors = {0,1};
    Solution S;
    cout << S.maxDistance(colors) << endl;
    return 0;
}

File: daily_python/2078_Two_Furthest_Houses_With_Different_Colors.py
import os
import sys
from typing import List

class Solution:
    def maxDistance(self, colors: List[int]) -> int:
        n = len(colors)
        if colors[0] != colors[-1]:
            return n - 1
        for i in range(1, n // 2 + 1):
            if colors[i] != colors[0] or colors[n - 1 - i] != colors[0]:
                return n - 1 - i
        return -1
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            colors = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            dataset.append(colors)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for colors in dataset:
        results.append(solution.maxDistance(colors))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')